### PR TITLE
Dm 411 emit connection status

### PR DESCRIPTION
**Pull request by dominikwerder**
### Issue

- Let forwarder emit EPICS connection status information in `ep00` schema
- Updates DM-411

### Description of work

- Output EPICS connection status in `ep00` format
- Add new CLI option `--service-id` analog to filewriter, emit in EPICS connection message
- Give class ChannelRequester its own file
- Let build nodes actually use the conan-built cmake
- Clean up cmake output paths
- Some LLVMification

### CODE REVIEW

**Comment by dominikwerder** [comment]
System test pipeline seems broken. Fails with

```
Recording test results
No test report files were found. Configuration error?
```

File: src/EpicsClient/ChannelRequester.cpp [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  // and that this requester callback is called from the same stack
+  // from which the channel creation was initiated.
+  LOG(Sev::Debug, "ChannelRequester::channelCreated:  (int)status.isOK(): {}",
+      (int)Status.isOK());
+  if (!Status.isOK() || !Status.isSuccess()) {
+    std::string ChannelName;
+    if (Channel) {
+      ChannelName = Channel->getChannelName();
+    }
+    std::ostringstream StringStream;
```
Comment (michele-brambilla) [inline src/EpicsClient/ChannelRequester.cpp]: Does the following make sense if Channel==nullptr ?

File: src/EpicsClient/ChannelRequester.cpp [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  default:
+    return ChannelConnectionState::UNKNOWN;
+  }
+}
+
+void ChannelRequester::channelStateChange(
+    Channel::shared_pointer const &Channel,
+    Channel::ConnectionState EpicsConnectionState) {
+  LOG(Sev::Debug, "channel state change: {}  for: {}",
+      connectionStateName(EpicsConnectionState), getChannelInfoString(Channel));
```
Comment (michele-brambilla) [inline src/EpicsClient/ChannelRequester.cpp]: What happens if Channel == nullptr?

File: src/EpicsClient/EpicsClientMonitor.cpp [hunk 3]
```diff
@@ -191,94 +72,55 @@ int EpicsClientMonitor::emitWithoutCaching(
...
+      toString(ConnectionState));
+  this->ConnectionState = ConnectionState;
+  if (ConnectionState == ChannelConnectionState::CONNECTED) {
+    Impl->monitoringStart();
+  } else {
+    Impl->monitoringStop();
+  }
+  if (ConnectionStatusProducer != nullptr) {
+    flatbuffers::FlatBufferBuilder Builder;
+    auto PVName = Builder.CreateString(Impl->channel_name);
```
Comment (michele-brambilla) [inline src/EpicsClient/EpicsClientMonitor.cpp]: This case can be implemented in a different method. In this way `handleConnectionStateChange` only detects the connection state and delegates other methods to react accordingly (like you do in the case of `ChannelConnectionState::CONNECTED`)

**Comment by mchorazak** [comment]
Closing because functionality was re-implemented on another branch.

**Pull request closed without merging**